compiler/src/iree/compiler/Codegen/LLVMGPU/TransformExtensions/LLVMGPUExtensions.cpp

```diff
@@ -1496,5 +1500,41 @@ void transform_dialect::PackSharedMemoryAllocOp::getEffects(
   transform::modifiesPayload(effects);
 }
 
+class TestVectorLayoutOptions : public VectorLayoutOptions {
+public:
+  TestVectorLayoutOptions(Operation *root) : VectorLayoutOptions(root) {}
+
+  void setAnchorOps(VectorLayoutAnalysis &analysis) override {
+    setAnchorOpsFromAttributes(analysis, root);
+  }
+};
+
+DiagnosedSilenceableFailure
+transform_dialect::TestAMDGPUContractionDistribution::applyToOne(
+    transform::TransformRewriter &rewriter, func::FuncOp target,
+    transform::ApplyToEachResultList &results,
+    transform::TransformState &state) {
+  TestVectorLayoutOptions options(target);
+  RewritePatternSet patterns(target.getContext());
+  int64_t m = getM();
+  int64_t n = getN();
+  int64_t k = getK();
+  MFMAType mfmaType;
+  if ((m == 16) && (n == 16) && (k == 16)) {
+    mfmaType = MFMAType::F16_16x16x16_F32;
+  } else {
+    mfmaType = MFMAType::F16_32x32x8_F32;
+  }
+  populateAMDGPUDistributionPatterns(patterns, mfmaType);
```

Comment:
Is it not possible for the patterns to match on the layout to check what mfmaType to use? I don't like passing mfmaType to patterns.

Reply:
Yes, I think this is possible.